src/utils.ts

```diff
@@ -26,16 +26,29 @@ function loadTsFile(filename: string, cwd: string, tsConfig?: any): any {
   let { data, path } = joycon.loadSync([filename], cwd)
   if (path && data) {
     if (tsConfig) {
-      data = deepmerge(tsConfig, data)
+      data = deepmerge(data, tsConfig)
     }
-    const { extends: _extends } = data
+    let { extends: _extends } = data
     if (_extends) {
       delete data.extends
-      return loadTsFile(path, cwd, data)
+      if(!_extends.endsWith('.json')){
+        _extends += '.json'
+      }
+      return loadTsFile(_extends, cwd, data)
     } else {
       return data
     }
   } else {
-    return null
+    return loadNodeModuleTsFile(filename, tsConfig)
```

Comment:
Naive "try and resolve from node module if not found". Once again could try and recurse for a completeness but much of a muchness.

Reply:
What if the tsconfig in node_module extends another one? I think we could extract this logic with `joycon.loadSync([filename], cwd)` together in a function, and put it in the beginning of this recursive function

Reply:
I was considering this approach actually. One hairy detail in that you currently do a `delete data.extends`, I'm wondering if it's necessary to do that delete at all.

Reply:
I can hardly tell why now as well... I think it may relate to the flipped merging logic. You could take it off to see if all tests still pass.

Reply:
nevermind delete would work fine.